crates/ty_server/src/server/api/notifications/did_open.rs

```diff
@@ -45,14 +47,23 @@ impl SyncNotificationHandler for DidOpenTextDocumentHandler {
         session.open_text_document(key.path(), document);
 
         let path = key.path();
+        let db = session.project_db_mut(path);
 
         match path {
             AnySystemPath::System(system_path) => {
+                match system_path_to_file(db, system_path) {
+                    Ok(file) => db.project().open_file(db, file),
+                    Err(err) => {
+                        // This can only fail when the path is a directory or it doesn't exists but the
+                        // file should exists for this handler in this branch.
```

Comment:
Let's explain why it should exist in this branch -> because it was added to the index, therefore the LSPSystem should return it when reading it

Reply:
Done